tests/sentry/ratelimits/test_redis.py

```diff
@@ -13,3 +15,34 @@ def test_project_key(self):
     def test_simple_key(self):
         assert not self.backend.is_limited("foo", 1)
         assert self.backend.is_limited("foo", 1)
+
+    def test_correct_current_value(self):
+        """Ensure that current_value get the correct value after the counter in incremented"""
+        for _ in range(10):
+            self.backend.is_limited("foo", 100)
+
+        assert self.backend.current_value("foo") == 10
+        self.backend.is_limited("foo", 100)
+        assert self.backend.current_value("foo") == 11
+
+    def test_current_value_new_key(self):
+        """current_value should return 0 for a new key"""
+
+        assert self.backend.current_value("new") == 0
+
+    def test_current_value_expire(self):
+        """Ensure that the count resets when the window expires"""
+        with freeze_time("2000-01-01") as frozen_time:
+            for _ in range(10):
+                self.backend.is_limited("foo", 1, window=1)
+
```

Comment:
Can you assert the value during the window is what you expect it to be as well? I know it's tested elsewhere but it'd make it more explicit if anyone reads through these later.

Reply:
This caught a bug. Nice catch!